feat: Event listener, docker support and development environment

ilhamfi27 opened the pull request:
Event listener can be processed and sent as a webhook Adding docker support for further development and scaling development

ilhamfi27 committed:
feat: Event listener, docker support and development environment
Event listener can be processed and sent as a webhook
Adding docker support for further development and scaling development

ilhamfi27 committed:
fix: removing secrets from dev docker container

dimaskiddo commented:
Hai @ilhamfi27 
First i want to thank you for your big contibution feature here 😄 

As note before i can merge it into `master` or `main` branch i think you need to PR to `develop` branch instead, because i need to adjust i few think more or may be you also can help with it 😉 

I want to adjust more like:
1. Remove any Logs Save to Database since i want to keep the compability with SQLite for easy Deployment, so it just Log it to the STDOUT
2. I want to Standardize the Webhook Call to the Endpoint like using POST method only with Known Well Structure lik
```
{
  "from": "<sender_msisdn_jid>",
  "to": "<reciever_msisdn_jid>", // This Make the Webhook Known Who is the Target if Multi Account is Used
  "text": "<text_or_caption>",
  "media": {
    "link": "<image_audio_document_media_link_to_download>"
  }
}
```
3. And I Want to Keep it Simple By just Adding the Webhook Activation function At Startup Routine and Webhook Event HTTP Call in WhatsApp Package

ilhamfi27 committed:
Merge remote-tracking branch 'original/develop'

ilhamfi27 committed:
refactor: remove database package and related code, simplify app initialization

ilhamfi27 committed:
Merge remote-tracking branch 'original/develop'

ilhamfi27 committed:
feat: implement webhook event handling and payload structure

- Added a new package for handling WhatsApp webhook events.
- Introduced WebhookPayload structure to standardize event data sent to the webhook.
- Implemented specific payload builders for various event types (message, receipt, presence, etc.).
- Enhanced app initialization to support a configurable whitelist of webhook events.
- Updated the WhatsApp event handler to construct and send payloads based on the event type.
- Removed unused imports and cleaned up code for better readability.

ilhamfi27 committed:
feat: add media storage support with local and S3 backends

ilhamfi27 committed:
feat: add webhook debugging functionality with JSON file output

ilhamfi27 committed:
feat: add endpoint to retrieve group information by Group ID

ilhamfi27 committed:
feat: add SendTyping endpoint and request structure for typing status
